source/grammar/qasm3Parser.g4

```diff
@@ -220,11 +220,17 @@ argumentDefinition:
     | (CREG | QREG) Identifier designator?
     | arrayReferenceType Identifier
 ;
+gateParameter:
```

Comment:
Perhaps `gateParameterList` should be a choice of `identifierList` or  list of typed identifier. This way the gate syntax can be validated at parse time rather than semantic analysis.